cg/meta/workflow/nf_analysis.py

```diff
@@ -307,3 +320,109 @@ def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
+        sample_id: str = case.links[0].sample.internal_id
+        multiqc_json = MultiqcDataJson(
+            **read_json(file_path=self.get_multiqc_json_path(case_id=case_id))
+        )
+        metrics_values: dict = {}
+
+        for key in multiqc_json.report_general_stats_data:
+            if case_id in key:
+                metrics_values.update(list(key.values())[0])
+                LOG.info(f"Key: {key}, Values: {list(key.values())[0]}")
+
+        return [
+            MetricsBase(
+                header=None,
+                id=sample_id,
+                input=MultiQC.MULTIQC_DATA + FileExtensions.JSON,
+                name=metric_name,
+                step=MultiQC.MULTIQC,
+                value=metric_value,
+                condition=self.get_pipeline_metrics().get(metric_name, None),
+            )
+            for metric_name, metric_value in metrics_values.items()
+        ]
+
+    def get_multiqc_json_metrics_per_sample(self, case_id: str) -> list[MetricsBase]:
+        """Get a multiqc_data.json file and returns metrics and values formatted for every sample_id."""
+        multiqc_json: MultiqcDataJson = MultiqcDataJson(
+            **read_json(file_path=self.get_multiqc_json_path(case_id=case_id))
+        )
+        samples: list = self.status_db.get_samples_by_case_id(case_id=case_id)
+        metrics_list: list[MetricsBase] = []
+        for sample in samples:
+            sample_id: str = sample.internal_id
+            metrics = self.get_metrics_from_sample(
+                sample_name=sample.name, multiqc_json=multiqc_json.report_general_stats_data
+            )
+            for metric_name, metric_value in metrics.items():
+                metrics_list.append(
+                    MetricsBase(
+                        header=None,
+                        id=sample_id,
+                        input=MultiQC.MULTIQC_DATA + FileExtensions.JSON,
+                        name=metric_name,
+                        step=MultiQC.MULTIQC,
+                        value=metric_value,
+                        condition=self.get_pipeline_metrics().get(metric_name, None),
+                    )
+                )
+
+        return metrics_list
+
+    @staticmethod
+    def get_metrics_from_sample(sample_name: str, multiqc_json: list[dict]) -> dict:
+        """Get tuple metric_name and metric_values from sample_id"""
+
+        metrics_values: dict = {}
+        for stat_dict in multiqc_json:
+            for sample_key, sample_values in stat_dict.items():
+                if sample_key == f"{sample_name}_{sample_name}":
+                    LOG.info(f"Key: {sample_key}, Values: {sample_values}")
+                    metrics_values.update(sample_values)
+
+        return metrics_values
+
+    def validate_qc_metrics(self, case_id: str, dry_run: bool = False) -> None:
+        """Validate the information from a QC metrics deliverable file."""
+
+        if dry_run:
```

Comment:
I will leave this as it is for now.

Reply:
I think putting the dry run after `MetricsDeliverablesCondition(**qc_metrics_raw)` will help in the development of a unit test, which this function has not

Reply:
Placing it right after `MetricsDeliverablesCondition` would make the `dry-run` useless if the metrics validation fails, so it's either keeping it there or moving it just before this model

Reply:
but isn't the point of dry run to test which parts will fail? I think what we want to avoid with the dry run is the last step, to set the analysis status in TB

Reply:
yes, but the thing is that the model validation is an expected fail and we don't want a dry-run to raise a ValidationError for QC metrics not passing their conditions

Reply:
I suggest we consider three options (assuming you agree, @diitaz93):

1. Leave it as is
2. Move it before the model initialisation
3. Skip only the TB calls and error raises
  - This has the benefit of distinguishing between QC fails and unexpected fails
  - It does make the code a bit more verbose, though

Feel free to implement the functionality that you find more convenient, @sofstam

Reply:
yes I agree, we can leave this discussion for later